Code/Framework/AzToolsFramework/AzToolsFramework/ContainerEntity/ContainerEntitySystemComponent.cpp

```diff
@@ -128,36 +128,32 @@ namespace AzToolsFramework
                 return entityId;
             }
 
-            // Return the highest closed container, or the entity if none is found.
-            AZ::EntityId secondLastOpenContainerBeforeSelection = entityId;
-            AZ::EntityId lastOpenContainerBeforeSelection = entityId;
+            // Return the highest unselected ancestor container of the entity, or the entity if none is found.
+            AZ::EntityId highestUnselectedAncestorContainer = entityId;
 
             // Skip the queried entity, as we only want to check its ancestors.
             AZ::TransformBus::EventResult(entityId, entityId, &AZ::TransformBus::Events::GetParentId);
 
-            // Assumption - there are no closed containers. This is due to the way PrefabEditScope::NESTED_INSTANCES works,
-            // since we open all descendant containers of the focused prefab.
-            // If this is no longer the case, this code will need a rewrite.
```

Comment:
question: Since we are removing these comments, would it be good to update the API comment to describe the "no closed container" case? It looks like if the override flag is on, the lines after 160 won't be executed at all.

```cpp
//! Detects if one of the ancestors of entityId is a closed container entity.
//! @return The highest closed entity container id if any, or entityId otherwise.
virtual AZ::EntityId FindHighestSelectableEntity(AZ::EntityId entityId) const = 0;
```

Reply:
I'm making a follow up ticket to change the logic so that there's no branching on PrefabsOverridesEnabled flag.